rskj-core/src/main/java/org/ethereum/core/ImmutableTransaction.java

```diff
@@ -0,0 +1,28 @@
+package org.ethereum.core;
+
+import org.ethereum.crypto.ECKey;
+
+/**
+ * Created by ajlopez on 02/08/2017.
+ */
+public class ImmutableTransaction extends Transaction {

```

Comment:
I think it would be best to have this reversed: `Transaction` inherits from `ImmutableTransaction` and adds the `sign` and `setGasLimit`. This way, instead of having exceptions thrown (i.e. runtime checks), we have the compiler check things for us (compile time checks).